### PR TITLE
Itt 584/automate services

### PR DESCRIPTION
**Pull request by wboereboom**
**Description**
This PR includes definition updates and automation for all service classes

**Breaking changes:**
- `Management` service renamed to `ManagementApi`. Subclass names changed to match camel-cased version of API Explorer description (e.g. `MyApiCredentialApi` and `AccountStoreLevelApi`)
- `StoredValue` service renamed to `StoredValueApi`. `mergebalance` endpoint function renamed to `mergeBalance` for consistency
- `BalancePlatform` service renamed to  `BalancePlatformApi`. Subclass names got suffixed with `Api` (e.g. `AccountHolders` becomes `AccountHoldersApi`
- `DataProtection` service renamed to `DataProtectionApi`

**Tested scenarios**
Through unit tests and manual testing

### CODE REVIEW

**Comment by jillingk** [comment]
Regenerated checkout with older specs, regenerated the rest with recent specs and updated some unit tests. Also added a line to set the config environment variable as merchant can also set the environment in the client. Lastly added two lines to throw an error whenever a live url needs to be set and no liveurlprefix is available.

**Review by jillingk — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -10,6 +10,7 @@ services:=balanceControl balancePlatform binlookup checkout configurationWebhook
 models: $(services)
 
 balanceControl: spec=BalanceControlService-v1
+balancePlatform: spec=BalancePlatformService-v2
```
Comment (michaelpaul) [inline Makefile:13]: Duplicate line, see line 24.

File: .github/workflows/services.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: install Node dev dependencies
+        run: npm install --save-dev
+      - run: make services
+      - name: Set PR variables
+        id: vars
+        run: |
+          cd build/spec
+          echo ::set-output name=pr_title::"Update models"
```
Comment (michaelpaul) [inline .github/workflows/services.yml:18]: Did you mean "Services"?

File: src/client.ts [hunk 3]
```diff
@@ -114,43 +78,15 @@ class Client {
     }
 
     public setEnvironment(environment: Environment, liveEndpointUrlPrefix?: string): void {
```
Comment (michaelpaul) [inline src/client.ts:80]: 'liveEndpointUrlPrefix' is defined but never used

File: src/service.ts [hunk 4]
```diff
@@ -26,6 +28,34 @@ class Service {
...
+        const config: Config = this.client.config;
+
+        if (config.environment !== "LIVE") {
+            return url;
+        }
+
+        if (url.includes("pal-")) {
+            if (this.client.liveEndpointUrlPrefix === "")
+            {
+                throw new Error("Please provide your unique live url prefix on the setEnvironment() call on the Client.")
```
Comment (michaelpaul) [inline src/service.ts:42]: Looks like `setEnvironment` is not the way to set live prefix anymore.

**Review by michaelpaul — CHANGES_REQUESTED** [review]

**Review by michaelpaul — APPROVED** [review]

**Pull request merged**